packages/mobile/src/components/Chat/ChatDetail.tsx

```diff
@@ -97,15 +99,16 @@ export const ChatDetail = memo(
     doOpenRetryModal,
     replyMessage,
     setReplyMessage,
-    assets,
-    setAssets,
     onPaste,
     hasMoreMessages,
     fetchMoreMessages,
     conversationId,
     onLinkData,
     onDismissLinkPreview,
+    onAudioRecord,
```

Comment:
Audio can be directly converted into an asset here right? No need to do that outside of this component.. If anything converting it into an asset should be deeper in the component tree, not higher up?

That way the components can be more generic the higher up they are, helping us add fancy new assets as some point, without having to change every component that touches it

Reply:
yess but whenever we record, we send it to chat. So onAudioRecord method contains more than asset conversion as it also handles sending the message directly to chat

Reply:
If it makes more sense to combine onAssetsAdded and onAudioRecord into one, I can make that work

Reply:
Yeah, I would definitely suggest to merge them into one, the fact that you directly send when it's audio, can still be handled based on the asset types from the chat page